libraries/Microsoft.Bot.Connector/Authentication/ManagedIdentityAppCredentials.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    /// <summary>
+    /// Managed Service Identity auth implementation.
+    /// </summary>
+    public class ManagedIdentityAppCredentials : AppCredentials
```

Comment:
Could this be internal to Microsoft.Bot.Builder.Integration.AspNet.Core or does it have uses outside of that library?

Reply:
@EricDahlvang - The existing password/certificate credential classes are public, so I followed the pattern to keep this managed credentials public as well. Also, the BuildAuthenticator() method is overridable, which developers can override if this is kept public.